docs/dataset.rst

```diff
@@ -0,0 +1,62 @@
+Dataset
+=======
+
+SQLDataset
+----------
+
+Creating a SQLDataset from a table in a db
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+:class:`~ml_tooling.data.SQLDataset` should be used for making datasets that get their data from a SQL based source.
+SQLDatasets must be provided with a :py:class:`sqlalchemy.engine.Connectable` or a valid connection string.
+
+
+.. code-block::
+
+    >>> import pandas as pd
+    >>> from ml_tooling.data import SQLDataset
+    >>> from sqlalchemy import create_engine
+    >>>
+    >>> class TableData(SQLDataset):
+    ...
+    ...     table = "TableName"
+    ...
+    ...     def load_training_data(self):
+    ...         return pd.read_sql_table(table, self.engine)
+    ...
+    ...     def load_prediction_data(self, idx):
+    ...         return pd.read_sql_table(table, self.engine).loc(idx)
+    >>>
+    >>> engine = create_engine("postgresql://username:password@localhost/test")
+    >>>
+    >>> TableData(engine)
+    <TableData - SQLDataset>
+
+
+FileDataset
+-----------
+
+Creating a FileDataset from a csv file
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Creating a FileDataset, one should define how the data is loaded based on the :code:`self.file_path` variable
```

Comment:
```suggestion
When we create a FileDataset, we need to specify the location of our datafiles - this will be available in the :code:`self.file_path` attribute
```